kmp-nativecoroutines-compiler/src/main/kotlin/com/rickclephas/kmp/nativecoroutines/compiler/diagnostics/KmpNativeCoroutinesErrors.kt

```diff
@@ -0,0 +1,68 @@
+package com.rickclephas.kmp.nativecoroutines.compiler.diagnostics
+
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.diagnostics.DiagnosticFactory0
+import org.jetbrains.kotlin.diagnostics.Errors
+import org.jetbrains.kotlin.diagnostics.Severity
+import org.jetbrains.kotlin.psi.KtElement
+
+internal object KmpNativeCoroutinesErrors {
+    @JvmField
+    val CONFLICT_COROUTINES = DiagnosticFactory0.create<KtElement>(Severity.ERROR)
+    @JvmField
+    val EXPOSED_FLOW_TYPE = DiagnosticFactory0.create<PsiElement>(Severity.WARNING)
+    @JvmField
+    val EXPOSED_FLOW_TYPE_ERROR = DiagnosticFactory0.create<PsiElement>(Severity.ERROR)
+    @JvmField
+    val EXPOSED_STATE_FLOW_PROPERTY = DiagnosticFactory0.create<PsiElement>(Severity.WARNING)
+    @JvmField
+    val EXPOSED_STATE_FLOW_PROPERTY_ERROR = DiagnosticFactory0.create<PsiElement>(Severity.ERROR)
+    @JvmField
+    val EXPOSED_SUSPEND_FUNCTION = DiagnosticFactory0.create<PsiElement>(Severity.WARNING)
+    @JvmField
+    val EXPOSED_SUSPEND_FUNCTION_ERROR = DiagnosticFactory0.create<PsiElement>(Severity.ERROR)
+    @JvmField
+    val IGNORED_COROUTINES = DiagnosticFactory0.create<KtElement>(Severity.ERROR)
+    @JvmField
+    val IGNORED_COROUTINES_REFINED = DiagnosticFactory0.create<KtElement>(Severity.ERROR)
+    @JvmField
+    val IGNORED_COROUTINES_REFINED_STATE = DiagnosticFactory0.create<KtElement>(Severity.ERROR)
+    @JvmField
+    val IGNORED_COROUTINES_STATE = DiagnosticFactory0.create<KtElement>(Severity.ERROR)
+    @JvmField
+    val INVALID_COROUTINES = DiagnosticFactory0.create<KtElement>(Severity.ERROR)
+    @JvmField
+    val INVALID_COROUTINES_IGNORE = DiagnosticFactory0.create<KtElement>(Severity.ERROR)
+    @JvmField
+    val INVALID_COROUTINES_REFINED = DiagnosticFactory0.create<KtElement>(Severity.ERROR)
+    @JvmField
+    val INVALID_COROUTINES_REFINED_STATE = DiagnosticFactory0.create<KtElement>(Severity.ERROR)
+    @JvmField
+    val INVALID_COROUTINES_STATE = DiagnosticFactory0.create<KtElement>(Severity.ERROR)
+    @JvmField
+    val INVALID_COROUTINE_SCOPE = DiagnosticFactory0.create<KtElement>(Severity.ERROR)
+    @JvmField
+    val REDUNDANT_OVERRIDE_COROUTINES = DiagnosticFactory0.create<KtElement>(Severity.ERROR)
```

Comment:
Should this really be an error?!

Reply:
At the moment it's actually a valid use case to annotate overrides, e.g. when implementing an interface.
However this isn't ideal so will convert these to a warning instead.